auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LdapRolesProvider.java

```diff
@@ -95,29 +140,54 @@ private void initialize() throws Exception {
 
         if (bindResult.getResultCode() != ResultCode.SUCCESS) {
             LOGGER.error("Error binding to LDAP" + bindResult.getResultCode());
-            throw new Exception("Error binding to LDAP" + bindResult.getResultCode());
+            throw new LDAPException(bindResult.getResultCode(), "Error binding to LDAP");
         }
 
         connectionPool = new LDAPConnectionPool(ldapConnection, ldapConfiguration.getPoolSize());
         connectionPool.setMaxConnectionAgeMillis(ldapConfiguration.getPoolMaxConnectionAgeMS());
-        LOGGER.info("Initialized LDAPConnectionPool: poolSize={}, poolMaxAge={}, connectionTimeout={}, responseTimeout={}, debug={}, keepAlive={}",
+        LOGGER.info("Initialized LDAPConnectionPool: poolSize={}, poolMaxAge={}, connectionTimeout={}, responseTimeout={}, debug={}, keepAlive={}.",
                 ldapConfiguration.getPoolSize(), ldapConfiguration.getPoolMaxConnectionAgeMS(), ldapConfiguration.getConnectionTimeoutMS(), ldapConfiguration.getResponseTimeoutMS(),
                 ldapConfiguration.isDebug(), ldapConfiguration.isKeepAlive());
 
     }
 
-    public SearchResult search(String baseDn, String filter) throws Exception {
+    @Override
+    public Set<String> getUserRoles(String username) throws Exception {
+        LOGGER.debug("getRoles("+username+")");
+
+        Objects.requireNonNull(username);
 
-        if (null == connectionPool) {
-            synchronized (LdapRepository.class) {
-                if(null == connectionPool) {
-                    initialize();
+        Set<String> roles = new HashSet<>();
+
+        String filter = "(uid=" + username + ")";
+
+        SearchRequest searchRequest = new SearchRequest(searchBase, SearchScope.SUB, filter);
+        SearchResult searchResult = connectionPool.search(searchRequest);
+        List<SearchResultEntry> searchResultEntries = searchResult.getSearchEntries();
+
+        if(searchResultEntries.isEmpty()) {
+            LOGGER.warn("No result found roles for user: " + username);
+            return new HashSet<String>();
+        } else if (searchResultEntries.size() > 1) {
```

Comment:
I don't even know if it's possible to happen (a uid= query matching more than one user), but if it can, it seems like an error. Authorization is sensitive, I don't want to introduce ambiguity by merging roles from all users who match.